lib/ecmascript.ts

```diff
@@ -6486,10 +6488,15 @@ export function CreateOnePropObject<K extends string, V>(propName: K, propValue:
   return o;
 }
 
-function CopyOptions<T extends { [s in K]?: unknown }, K extends string & keyof T>(options: T | undefined) {
-  const optionsCopy = ObjectCreate(null) as NonNullable<T>;
-  CopyDataProperties(optionsCopy, GetOptionsObject(options), []);
-  return optionsCopy;
+export function SnapshotOwnProperties<T extends { [s in K]?: unknown }, K extends string & keyof T>(
```

Comment:
Should k just be PropertyKey?

Reply:
I replaced `string` above with `PropertyKey` and didn't see any ill effects, so I checked in that. The `& keyof T` is required though. Without it, a bunch of TS errors come up.